executor/api/rest/client_test.go

```diff
@@ -266,3 +269,32 @@ func TestClientMetrics(t *testing.T) {
 	}
 
 }
+
+func TestErrorResponse(t *testing.T) {
+	t.Logf("Started")
+	g := NewGomegaWithT(t)
+	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(errorPredictResponse))
+	})
+	host, port, httpClient, teardown := testingHTTPClient(g, h)
```

Comment:
Gomega has its own package for testing http clients `ghttp`: https://onsi.github.io/gomega/#ghttp-testing-http-clients

Might be worth looking into seeing as we're using it for assertions.

Reply:
I think `ghttp` is not compatible with `NewGomegaWithT` yet (https://github.com/onsi/gomega/issues/321) as it's mainly designed to be used with `gingkgo`. However, given that we already have some manual `teardown()`, it could make sense to start using `ginkgo` (and `ghttp`).

Reply:
Maybe we create an issue for this?